File: lang/java/aingle/src/main/java/org/apache/aingle/file/NullCodec.java
/*

 */
package org.apache.aingle.file;

import java.io.IOException;
import java.nio.ByteBuffer;

/** Implements "null" (pass through) codec. */
final class NullCodec extends Codec {

  private static final NullCodec INSTANCE = new NullCodec();

  static class Option extends CodecFactory {
    @Override
    protected Codec createInstance() {
      return INSTANCE;
    }
  }

  /** No options available for NullCodec. */
  public static final CodecFactory OPTION = new Option();

  @Override
  public String getName() {
    return DataFileConstants.NULL_CODEC;
  }

  @Override
  public ByteBuffer compress(ByteBuffer buffer) throws IOException {
    return buffer;
  }

  @Override
  public ByteBuffer decompress(ByteBuffer data) throws IOException {
    return data;
  }

  @Override
  public boolean equals(Object other) {
    if (this == other)
      return true;
    return (other != null && other.getClass() == getClass());
  }

  @Override
  public int hashCode() {
    return 2;
  }
}

File: lang/csharp/src/apache/main/File/NullCodec.cs
/*

 */

namespace AIngle.File
{
    /// <summary>
    /// Implements a codec that does not perform any compression. This codec simply returns the
    /// bytes presented to it "as-is".
    /// </summary>
    public class NullCodec : Codec
    {
        /// <summary>
        /// Initializes a new instance of the <see cref="NullCodec"/> class.
        /// </summary>
        public NullCodec() { }

        /// <inheritdoc/>
        public override byte[] Compress(byte[] uncompressedData)
        {
            return uncompressedData;
        }

        /// <inheritdoc/>
        public override byte[] Decompress(byte[] compressedData)
        {
            return compressedData;
        }

        /// <inheritdoc/>
        public override string GetName()
        {
            return DataFileConstants.NullCodec;
        }

        /// <inheritdoc/>
        public override bool Equals(object other)
        {
            if (this == other)
                return true;
            return this.GetType().Name == other.GetType().Name;
        }

        /// <inheritdoc/>
        public override int GetHashCode()
        {
            return DataFileConstants.NullCodecHash;
        }
    }
}
